src/components/rate/Rate.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+    <div class="rate" :class="{ 'is-disabled': disabled, 'is-spaced': spaced }">
+        <div
+            v-for="(item, index) in max"
+            class="rate-item"
+            :class="rateClass(item)"
+            :key="index"
+            @mousemove="previewRate(item, $event)"
+            @mouseleave="resetNewValue"
+            @click.prevent="confirmValue(item)">
+            <b-icon
+                :pack="iconPack"
+                :icon="icon"
+                :size="size"
+            />
+            <b-icon
+                v-if="checkHalf(item)"
+                class="is-half"
+                :pack="iconPack"
+                :icon="icon"
+                :size="size"
+                :style="halfStyle"
+            />
+        </div>
+        <div
+            class="rate-text"
+            :class="size"
+            v-if="showText || showScore || customText">
+            <span>{{ showMe }}</span>
+            <span v-if="customText && !showText">{{ customText }}</span>
+        </div>
+    </div>
+</template>
+
+<script>
+import Icon from '../icon/Icon'
+
+export default {
+    name: 'BRate',
+    components: {
+        [Icon.name]: Icon
+    },
+    props: {
+        value: {
+            type: Number,
+            default: 0
+        },
+        max: {
+            type: Number,
+            default: 5
+        },
+        icon: {
+            type: String,
+            default: 'star'
+        },
+        iconPack: String,
+        size: String,
+        spaced: Boolean,
+        disabled: Boolean,
+        showScore: Boolean,
+        showText: Boolean,
+        customText: String,
+        texts: {
+            type: Array,
+            default() {
+                return ['Very Bad', 'Bad', 'Good', 'Very Good', 'Awesome']
```

Comment:
I'm not sure we should include default texts. It would be too much opiniated I think.

Reply:
What do you think about having default texts @jtommy ?

Reply:
@apolokaklab  I agree with @service-paradis , please remove it

Reply:
Could add an example in the doc on how to use it :)